qiskit/transpiler/passes/scheduling/padding.py

```diff
@@ -205,11 +258,13 @@ def _pad(
         self,
         dag: DAGCircuit,
         qubit: Qubit,
-        time_interval: int,
+        t_start: int,
+        t_end: int,
         next_node: DAGNode,
         prev_node: DAGNode,
     ):
         if not self.fill_very_end and isinstance(next_node, DAGOutNode):
             return
 
-        dag.apply_operation_back(Delay(time_interval, dag.unit), [qubit])
+        time_interval = t_end - t_start
+        self._apply_scheduled_op(dag, t_start, Delay(time_interval, dag.unit), qubit)
```

Comment:
does it raise an error if you try to insert a longer instruction? for example if I try to do `Delay(2 * time_interval)` here what happens?

Reply:
Good point. Unfortunately I have no idea to detect this error. It just creates unexpected circuit, and extra delay of `1 * time_interval` will be scheduled on the backend compiler and the circuit is executed normally. We can let the `_pad` method create the new `DAGCircuit` and let the base padding `compose` the returned DAG with duration evaluation, but it looks like unnecessary overhead. DD pass guarantees the total spacing becomes 1, thus this situation doesn't happen on user end.

Reply:
ok if it's not a pitfall for the user let's let it stay but mabe add a comment to that function that it does not check for validity of the op.

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/7760/commits/095274ba6df421bc63308f5a9eec0543d984f189